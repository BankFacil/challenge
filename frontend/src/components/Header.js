import React from 'react';

const Header = () => (
    <header className="header">
        <div className="header__container">
            <h1 className="header__title">
                <a href="#" title="Creditas">
                    <img className="logo" src="https://staging.creditas.com.br/static/images/logo-creditas-white-3cd22a2808.svg" alt="Creditas" />
                </a>
            </h1>
            <a className="help" href="#ajuda">Ajuda</a>
        </div>
    </header>
);

export default Header;