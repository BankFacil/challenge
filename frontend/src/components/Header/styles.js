import styled from "styled-components";
import { colors } from "../../styles/index";

export const Wrapper = styled.header`
  background-color: ${colors.mountainMeadow};
  padding: 20px 0;
`;

export const Container = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 80%;
  margin: 0 auto;
  a {
    text-decoration: none;
    color: ${colors.concrete};
  }
`;

export const Title = styled.h1``;
export const Logo = styled.img`
  height: 1em;
`;
