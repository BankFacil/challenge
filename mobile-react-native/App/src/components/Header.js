import React from 'react';
import { Box, Image } from '.';

import LogoLabel from '../../assets/github-label.png';

const Header = () => (
	<Box
		alignItems="center"
		justifyContent="center"
		backgroundColor="white"
		height={45}
		>
		<Image source={LogoLabel} height={40} resizeMode="center" />
	</Box>
);

export default Header;