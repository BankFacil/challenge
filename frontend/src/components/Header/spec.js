/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-undef */
import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { getByTestId } from '@testing-library/dom';
import { act } from 'react-dom/test-utils';

import Header from './Header';

let container = null;
beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

it('components/Header', () => {
    act(() => {
        render(
            <Header />,
            container,
        );
    });

    const header = getByTestId(container, 'header');

    const expectValue = 'Ajuda';
    expect(header.textContent).toBe(expectValue);
});
