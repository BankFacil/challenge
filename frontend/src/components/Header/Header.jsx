import React from 'react';
import styles from './styles';

const Header = () => (
    <header {...styles} data-testid="header">
        <div className="header__container">
            <h1 className="header__title">
                <a href="#home" title="Creditas">
                    <img className="logo" src="https://staging.creditas.com.br/static/images/logo-creditas-white-3cd22a2808.svg" alt="Creditas" />
                </a>
            </h1>
            <a className="help" href="#ajuda">Ajuda</a>
        </div>
    </header>
);

export default Header;
