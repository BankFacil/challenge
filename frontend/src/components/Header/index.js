import React from "react";
import { Link } from "react-router-dom";

import { Wrapper, Container, Title, Logo } from "./styles";

export default function Header() {
  return (
    <Wrapper>
      <Container>
        <Title>
          <Link to="/">
            <Logo
              className="logo"
              src="https://staging.creditas.com.br/static/images/logo-creditas-white-3cd22a2808.svg"
              alt="Creditas"
            />
          </Link>
        </Title>
        <Link to="/help">Ajuda</Link>
      </Container>
    </Wrapper>
  );
}
