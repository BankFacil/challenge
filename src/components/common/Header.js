import React from 'react';
import PropTypes from 'prop-types';

import HeaderLogoImg from '../../assets/images/HeaderLogoImg.png';
import {HeaderContainer, HeaderLogo} from './styles';
import BackIcon from './BackIcon';

const Header = ({onBack}) => (
  <HeaderContainer>
    {onBack && <BackIcon onPress={onBack} />}
    <HeaderLogo resizeMode={'contain'} source={HeaderLogoImg} />
  </HeaderContainer>
);

Header.propTypes = {
  onBack: PropTypes.func,
};

export default Header;
