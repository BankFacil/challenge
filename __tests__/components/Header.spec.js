import React from 'react';
import renderer from 'react-test-renderer';

import Header from '../../src/components/common/Header';

describe('Header', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const tree = renderer.create(<Header />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should match to snapshot', () => {
      const tree = renderer.create(<Header onBack={() => {}} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
