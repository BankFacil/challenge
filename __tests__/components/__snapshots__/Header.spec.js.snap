// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Rendering should match to snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "paddingBottom": 6,
        "paddingLeft": 6,
        "paddingRight": 6,
        "paddingTop": 6,
        "width": "100%",
      },
    ]
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../src/assets/images/HeaderLogoImg.png",
      }
    }
    style={
      Array [
        Object {
          "height": 45,
        },
      ]
    }
  />
</View>
`;

exports[`Header Rendering should match to snapshot 2`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "paddingBottom": 6,
        "paddingLeft": 6,
        "paddingRight": 6,
        "paddingTop": 6,
        "width": "100%",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": 12,
        "opacity": 1,
        "position": "absolute",
        "top": 16,
        "zIndex": 99,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/images/BackIconImg.png",
        }
      }
      style={
        Array [
          Object {
            "height": 25,
            "width": 25,
          },
        ]
      }
    />
  </View>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../src/assets/images/HeaderLogoImg.png",
      }
    }
    style={
      Array [
        Object {
          "height": 45,
        },
      ]
    }
  />
</View>
`;
